textworld/gym/envs/batch_env.py

```diff
@@ -0,0 +1,297 @@
+
+import multiprocessing as mp
+from typing import Tuple, List, Dict
+
+import numpy as np
+
+import gym
+
+
+def _child(id, pipe):
+    """
+    Event loop run by the child processes
+    """
+    env = gym.make(id)
+
+    try:
+        while True:
+            command = pipe.recv()
+            # command is a tuple like ("call" | "get", "name.of.attr", extra args...)
+
+            obj = env
+            attrs = command[1].split(".")
+            for attr in attrs[:-1]:
+                obj = getattr(obj, attr)
+
+            if command[0] == "call":
+                fct = getattr(obj, attrs[-1])
+                result = fct(*command[2])
+            elif command[0] == "get":
+                result = getattr(obj, attrs[-1])
+            elif command[0] == "hasattr":
+                result = hasattr(obj, attrs[-1])
+
+            pipe.send(result)
+    finally:
+        pipe.close()
+        print(id, "closed")
+
+
+class _ChildEnv:
+    """
+    Wrapper for an env in a child process.
+    """
+    def __init__(self, id):
+        self._pipe, child_pipe = mp.Pipe()
+        self._process = mp.Process(target=_child, args=(id, child_pipe))
+        self._process.daemon = True
```

Comment:
@tavianator I added this to make sure the children get killed if the parent process dies.